### PR TITLE
[SPARK-4134] [core] Lower severity of some executor loss logs.

### PR DESCRIPTION
**Pull request by vanzin**
Don't log ERROR messages when executors are explicitly killed or when
the exit reason is not yet known.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala [hunk 1]
```diff
@@ -492,6 +492,18 @@ private[spark] class TaskSchedulerImpl(
...
+  private def logExecutorLoss(
+      executorId: String,
+      hostPort: String,
+      reason: ExecutorLossReason): Unit = reason match {
+    case LossReasonPending =>
+      logDebug(s"Executor $executorId on $hostPort lost, but reason not yet known.")
+    case ExecutorKilled(message) =>
+      logInfo(s"Executor $executorId on $hostPort killed by driver ($message)")
+    case _ =>
+      logError("Lost executor %s on %s: %s".format(executorId, hostPort, reason))
```
Comment (jerryshao) [inline core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala]: Is it better to unify to use string interpolation?

Comment (vanzin) [inline core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala]: Sure.

**Comment by jerryshao** [comment]
LGTM, indeed a nice improvement :).

**Comment by vanzin** [comment]
retest this please

**Comment by vanzin** [comment]
hmm unrelated failure (flaky test maybe, need to take a look). retest this please

**Comment by andrewor14** [comment]
@vanzin does this overlap with https://issues.apache.org/jira/browse/SPARK-4134?

**Comment by andrewor14** [comment]
retest this please

**Comment by vanzin** [comment]
@andrewor14 yes, I was actually looking for that bug instead of filing a new one, but couldn't find it. I'll update things.

File: core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala [hunk 2]
```diff
@@ -488,6 +493,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
       doRequestTotalExecutors(
         numExistingExecutors + numPendingExecutors - executorsPendingToRemove.size)
     } else {
+
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala]: delete

File: core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala [hunk 3]
```diff
@@ -474,12 +480,11 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
       logWarning(s"Executor to kill $id does not exist!")
     }
 
-    // If an executor is already pending to be removed, do not kill it again (SPARK-9795)
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala]: need to keep this?

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala [hunk 4]
```diff
@@ -493,7 +493,7 @@ private[yarn] class YarnAllocator(
         } else {
           logInfo(containerExitReason)
         }
-        ExecutorExited(0, exitCausedByApp, containerExitReason)
+        ExecutorExited(exitStatus, exitCausedByApp, containerExitReason)
```
Comment (andrewor14) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala]: this is unrelated right? just wanted to double check but this fix looks OK

**Comment by andrewor14** [comment]
This change looks fine. Thanks for fixing it.

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala (hunk 4 above)
Comment (vanzin) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala]: yeah, unrelated. just noticed it while reading that code.

**Comment by andrewor14** [comment]
Merging into master 1.6.

**Pull request closed without merging**